tests/pipelines/test_pipelines_image_segmentation.py

```diff
@@ -259,25 +266,43 @@ def test_small_model_pt(self):
         # The `panoptic` returns only LABEL_215, and this returns 3 labels.
         #
         output = image_segmenter("http://images.cocodataset.org/val2017/000000039769.jpg", subtask="semantic")
+
+        output_masks = [o["mask"] for o in output]
+        expected_masks = [
+            "https://huggingface.co/datasets/hf-internal-testing/mask-for-image-segmentation-tests/resolve/main/mask_0.png",
+            "https://huggingface.co/datasets/hf-internal-testing/mask-for-image-segmentation-tests/resolve/main/mask_1.png",
+            "https://huggingface.co/datasets/hf-internal-testing/mask-for-image-segmentation-tests/resolve/main/mask_2.png",
+        ]
+        expected_masks = [Image.open(requests.get(image, stream=True).raw) for image in expected_masks]
+
+        for output_mask, expected_mask in zip(output_masks, expected_masks):
```

Comment:
Can we remove the zipping ?It makes things easier to read imo:

```python
self.assertEqual(np.mean(outputs[0]["mask"], expected_mask[0]), ....)
self.assertEqual(outputs[1]["mask"], expected_mask[1])
self.assertEqual(outputs[2]["mask"], expected_mask[2])
```

And putting these tests *after* `self.assertEqual(nested_simplify(..) , ...)` (because that test will naturally check for output lengths, and is "easier" to pass than this one.

Reply:
Zipping in tests is adding indirection, which makes reasoning about them harder. 
If either list is empty, the `zip` is empty and the tests pass despite being extremely wrong (that's 1 example, but even more subtle things can creep) !

Reply:
I can do that as there is only 3 items to check here. But I think if we check the length of the 2 lists first, there shouldn't be creep thing happening. Zipping is used in other places of the testing, for example, check the PT/TF outputs are close enough to each other.

Let me know if you still feel strong of not using zip.

Reply:
I feel strongly about zipping in tests. I have seen incorrectly passing tests in `transformers` before.
In general nested code (for loops, ifs) are not desirable.
Sometimes a necessary evil, but in general it's better iMHO if we can avoid them.